File: CoreAutomata/Size.cs
using System;

namespace CoreAutomata
{
    public struct Size
    {
        public Size(int Width, int Height)
        {
            this.Width = Width;
            this.Height = Height;
        }

        public int Width { get; set; }

        public int Height { get; set; }

        public static Size operator *(Size Size, double Scale)
        {
            return new Size((Size.Width * Scale).Round(), (Size.Height * Scale).Round());
        }

        public override string ToString() => $"{Width}x{Height}";
    }
}

File: libautomata/src/main/java/com/mathewsachin/libautomata/Size.kt
package com.mathewsachin.libautomata

import kotlin.math.roundToInt

/**
 * A class for storing the [width] and [height] of something.
 */
data class Size(val width: Int, val height: Int) {
    init {
        require(width > 0) { "width must be positive" }
        require(height > 0) { "height must be positive" }
    }

    /**
     * Returns a new [Size] where X and Y were multiplied with [scale].
     */
    operator fun times(scale: Double): Size {
        return Size(
            (width * scale).roundToInt(),
            (height * scale).roundToInt()
        )
    }
}